models/users.js

```diff
@@ -483,6 +483,34 @@ const getDiscordUsers = async () => {
   }
 };
 
+const updateRoles = async (userData, newRoles) => {
+  try {
+    const roles = { ...userData.roles };
+    if (Object.keys(newRoles).includes("member")) {
+      if (newRoles.member === true) {
+        if (userData.roles.member) {
+          return { isRoleUpdated: false };
+        }
+      } else if (newRoles.member === false) {
+        if (userData.roles.member === false) {
+          return { isRoleUpdated: false };
+        }
+      }
+      roles.member = newRoles.member;
+    } else if (Object.keys(newRoles).includes("archived")) {
+      if (userData.roles && userData.roles.archived) return { isRoleUpdated: false };
```

Comment:
what if `userData.roles.archived` is false and user wants to change to false . you have only handled the true case. update the test case as well.

Reply:
If you see in in userModel database...archieved: false is already present in roles when user signup
Once I have to talk to @vinayak-trivedi...should we handle case once user get archieved...Does we handle case to user archieved: true to false also

Thanks for pointing

Reply:
Hi @heyrandhir, Write optimized code in my latest commit...can you please check